airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -201,7 +199,13 @@ private Throwable getRootConfigError(final Exception e) {
   }
 
   private boolean isConfigError(final Throwable e) {
-    return e instanceof ConfigErrorException || e instanceof ConnectionErrorException;
+    return e instanceof ConfigErrorException || e instanceof ConnectionErrorException || isRecoveryConnectionException(e);
+  }
+
+  private boolean isRecoveryConnectionException(Throwable e) {
```

Comment:
Seeing as we will be adding more logic to this class - perhaps we can move it to a `ConnectorExceptionUtil.java` class. WDYT?